### ISSUE TITLE
Version 2.1.1 doesn’t work with Fennec

### ISSUE THREAD

**Issue by jtotht** [M-Reimer/android-pdf-js#16]
# Steps to reproduce
1. Install the latest Fennec version, Firefox for Android 68.11.0 (available from the [FTP archives](https://ftp.mozilla.org/pub/mobile/releases/68.11.0/)).
2. Install Android PDF.js 2.1.1.
3. Open any PDF, e.g. https://www.ctan.org/tex-archive/macros/latex/contrib/beamer/doc/beameruserguide.pdf

# Actual result
4. An error message appears:
```
PDF.js v2.6.347 (build: 3be9c65f)
Message: The browser/environment lacks native support for critical functionality used by the PDF.js library (e.g. `ReadableStream` and/or `Promise.allSettled`); please use an ES5-compatible build instead.
```

# Expected result
4. The PDF loads.

It’s okay if you say you don’t want to work on making this version compatible with an end-of-life browser, but then please mark it as such in AMO so that Fennec doesn’t update to it automatically again and again. Of course the best solution would be making it compatible.

By the way, exactly your extension’s unavailability for Fenix stops me upgrading to it, simply I can’t live without it. A big thanks!

**Comment by M-Reimer** [comment]
It is not recommended or even supported to use an outdated browser. Fennec has not been updated for years which means serious security risks if you still use it!

The current "Android PDF.js" is meant to be used with "Firefox for Android Nightly" which does allow you to use other Add-ons.

To also give some sign to Mozilla, currently my recommended way to use Firefox on Android is to uninstall the "regular" Firefox (which also removes one installation from Mozilla's statistics) and only install the "Nightly" version.

**Comment by jtotht** [comment]
> It is not recommended or even supported to use an outdated browser.

Of course Mozilla doesn’t support using an outdated browser, this is why it’s called “unsupported”. Extension authors, however, are free decide what versions they support. This is why the [`browser_specific_settings`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/browser_specific_settings) manifest key exists, which allows you to declare what versions you actually support. If you don’t declare what versions you support, AMO assumes all versions from 42 up, which is obviously not the case here.

> Fennec has not been updated for years

10 months, see the timestamps in the FTP directory I linked. (Or do I have issues with terminology? I thought the last v68 version released in last July was still internally named Fennec.) I know it’s still a security risk, but switching to nightly is not an option for me for various reasons.

**Comment by M-Reimer** [comment]
I use nightly since about 6 months. It is extremely stable. I'll reconsider this but most probably wontfix

**Comment by M-Reimer** [comment]
Would be great if someone could create a pull request for this. I probably won't find time to fix this in the near future.

**Comment by Bardock88** [comment]
> 10 months, see the timestamps in the FTP directory I linked. (Or do I have issues with terminology? I thought the last v68 version released in last July was still internally named Fennec.) I know it’s still a security risk, but switching to nightly is not an option for me for various reasons.

If you want to use Fennec, then go with Fennec F-Droid, which looks like the old project with a new look: you can use any extension in a custom list, it has about:config and no telemetry.
There is also another excellent option that uses what Fennec F-Droid does and integrates the arkefox and tor uplift project into one, which is called Mull.

**Comment by M-Reimer** [comment]
> If you want to use Fennec, then go with Fennec F-Droid, which looks like the old project with a new look: you can use any extension in a custom list, it has about:config and no telemetry.

This may actually be a pretty interesting solution. But this is NOT "Fennec". Even the f-droid page states that it is Fenix but with some modifications. I'll keep this in mind to try this at some point.

And to finally get rid of this issue I've set the minimum required version to version 70. To get this to AMO it requires a new version to be released but at least on GIT it is now added.
